Providers/Profile.swift

```diff
@@ -903,13 +906,21 @@ public class BrowserProfile: Profile {
         }
 
         func syncEverything() -> Success {
-            return self.syncSeveral(
+            let job = self.syncSeveral(
```

Comment:
The is-sync-in-progress behavior either needs to end up inside `syncSeveral` or `withSyncInputs`, or every caller of either of those functions needs to check for the current sync; we don't want only one spot (`syncEverything`) to be checking.

Notably there are a few spots where we run partial syncs for a specific purpose: for example, `syncClientsThenTabs`. We might choose to use a `Deferred<Maybe<[EngineIdentifier]>>` as `currentSync` in order to identify what's currently syncing; when that sync is done, we'll know whether it synced what we wanted to sync, and if it didn't we can queue up a partial sync for the rest.